lib/ex_doc/autolink.ex

```diff
@@ -121,52 +122,52 @@ defmodule ExDoc.Autolink do
   end
 
   defp app(module) do
-    {_, app} = app_info(module)
-    app
+    case :code.which(module) do
+      :preloaded ->
+        :erts
+
+      maybe_path ->
+        case :application.get_application(module) do
+          {:ok, app} ->
+            app
+
+          _ ->
+            with true <- is_list(maybe_path),
+                 [_, "ebin", app, "lib" | _] <- maybe_path |> Path.split() |> Enum.reverse() do
+              String.split(app, "-") |> Enum.at(0) |> String.to_atom()
+            else
+              _ -> nil
+            end
+        end
+    end
   end
 
   @doc false
   def tool(module, config) do
     if match?("Elixir." <> _, Atom.to_string(module)) do
       :ex_doc
     else
-      {otp, app} = app_info(module)
-      apps = Enum.uniq(config.apps ++ Keyword.keys(config.deps))
-
-      if otp == true and app not in apps do
-        :otp
-      else
-        :ex_doc
-      end
+      tool_app(app(module), config)
     end
   end
 
-  defp app_info(module) do
-    case :code.which(module) do
-      :preloaded ->
-        {true, :erts}
+  defp tool_app(app, config) do
+    apps = Enum.uniq(config.apps ++ Keyword.keys(config.deps))
 
-      maybe_path ->
-        otp? = is_list(maybe_path) and List.starts_with?(maybe_path, :code.lib_dir())
-
-        app =
-          case :application.get_application(module) do
-            {:ok, app} ->
-              app
-
-            _ ->
-              with true <- is_list(maybe_path),
-                   [_, "ebin", app, "lib" | _] <- maybe_path |> Path.split() |> Enum.reverse() do
-                String.split(app, "-") |> Enum.at(0) |> String.to_atom()
-              else
-                _ -> nil
-              end
-          end
+    cond do
+      is_app_otp(app) and app not in apps ->
```

Comment:
```suggestion
      is_app_otp(app) and app not in config.apps  ->
```

I can't think of when an otp app would be in deps. :) This way we can get rid of the `apps` variable.

Reply:
When I build the docs for Erlang/OTP, otp apps are in deps :)

Reply:
@garazdawi then wouldn't the tool accuse to be ExDoc and generate links for HexDocs? Or maybe it does not matter because deps will use their own URLs anyway, so we can apply this change anyway?